packages/cli/src/cog/cutline.ts

```diff
@@ -178,42 +115,89 @@ export class Cutline {
      * Generate an optimized WebMercator tile cover for the supplied polygons
      * @param featureCollection Source TIff Polygons in GeoJson WGS84
      */
-    optimizeCovering(sourceMetadata: SourceMetadata): QuadKeyTrie {
-        const sourceZ = sourceMetadata.resolution;
+    optimizeCovering(sourceMetadata: SourceMetadata): string[] {
+        const srcArea = this.findCovering(sourceMetadata.bounds, this.polygons);
+        if (this.polygons.length !== 0) this.polygons = srcArea;
 
-        // Don't make COGs with a quadKey shorter than minZ.
+        const sourceZ = sourceMetadata.resolution;
         const minZ = Math.max(1, sourceZ + ZoomDifferenceForMaxImage);
+        // Don't make COGs with a quadKey shorter than minZ.
 
-        const covering = this.findCovering(sourceMetadata);
-
-        covering.mergeQuadKeys(CoveringPercentage, minZ, minZ + 2);
-
-        /** We should never return a full cover, using '' as a index causes problems */
-        if (covering.has('')) return QuadKeyTrie.fromList(QuadKey.children(''));
-
-        return covering;
+        const { qks } = this.makeQuadKeys('', srcArea, minZ, CoveringPercentage);
+        return qks.sort(QuadKey.compareKeys);
     }
 
     /**
      * Convert JobCutline to geojson FeatureCollection
      */
     toGeoJson(): FeatureCollection {
-        const coords: Position[][][] = [];
-        for (const poly of this.polygons.values()) {
-            coords.push([poly]);
-        }
+        const { forward } = TmsUtil.getProjection(this.targetTms);
+        return GeoJson.toFeatureCollection([
+            GeoJson.toFeatureMultiPolygon(this.polygons.map((p) => [p[0].map((q) => forward(q))])),
+        ]);
+    }
 
-        return GeoJson.toFeatureCollection([GeoJson.toFeatureMultiPolygon(coords)]);
+    /**
+     * Merge child nodes that have at least a covering fraction (`cf`)
+     * @param quadKey the quadKey to descend
+     * @param srcArea the aread of interest
+     * @param minZ Don't produce any quadKeys of this length or less and creater than this length + 2
+     * @param cf merge tiles that cover at least this fraction
+     * @return the quadKeys and percentage covered of the world by this srcArea set
+     */
+    private makeQuadKeys(
+        quadKey: string,
+        srcArea: MultiPolygon,
+        minZ: number,
+        cf: number,
+    ): { qks: string[]; af: number } {
```

Comment:
could we be more descriptive? `quadKeys` `coveringFactor` 

what is a `af` ?

Reply:
af => actualFraction